### PR TITLE
Adds upgradability to L2 Migrator contract 

### PR DESCRIPTION
**Pull request by kautukkundan**
**What does this pull request do? Explain your changes. (required)**

Adds upgradability to L2 Migrator contract in case the contract needs to be updated. Using the same ManagerProxy and ManagerProxyTarget pattern used by [protocol contracts ](https://github.com/livepeer/protocol/blob/streamflow/contracts/ManagerProxy.sol).

**Specific updates (required)**

- Copied contracts from protocol repo 
- Updated L2 Migrator to use ManagerProxyTarget
- Updated unit tests 

**How did you test each of these updates (required)**

ran unit tests

**Does this pull request close any open issues?**

**Checklist:**

- [x] README and other documentation updated
- [x] All tests using `yarn test` pass

### CODE REVIEW

**Review by yondonfu — CHANGES_REQUESTED** [review]
Coverage is down with the introduce of the contracts in the `proxy` folder. Since the tests for those contracts are already in the protocol repo we can ignore them in the `.solcover.js` file since the code coverage is addressed elsewhere.

File: contracts/L2/gateway/L2Migrator.sol [hunk 1]
```diff
@@ -41,10 +40,10 @@ interface IDelegatorPool {
     function claim(address _addr, uint256 _stake) external;
 }
 
-contract L2Migrator is L2ArbitrumMessenger, IMigrator, AccessControl {
-    address public immutable bondingManagerAddr;
-    address public immutable ticketBrokerAddr;
-    address public immutable merkleSnapshotAddr;
+contract L2Migrator is ManagerProxyTarget, L2ArbitrumMessenger, IMigrator {
+    address public bondingManagerAddr;
```
[the remark this answers is not in the document]
Comment (kautukkundan) [reply contracts/L2/gateway/L2Migrator.sol:44]: tracking here https://github.com/livepeer/arbitrum-lpt-bridge/issues/65

File: contracts/L2/gateway/L2Migrator.sol [hunk 2]
```diff
@@ -77,15 +76,21 @@ contract L2Migrator is L2ArbitrumMessenger, IMigrator, AccessControl {
         uint256 fees
     );
 
-    constructor(
+    /**
+     * @notice L2Migrator constructor. Only invokes constructor of base Manager contract with provided Controller address
+     * @dev This constructor will not initialize any state variables besides `controller`.
+     * - initialize() function must be called immediately after deployment
```
[the remark this answers is not in the document]
Comment (kautukkundan) [reply contracts/L2/gateway/L2Migrator.sol]: fixed in 010e074

**Review by yondonfu — APPROVED** [review]
LGTM!

**Pull request merged**